pkg/webhooks/hiveextension/v1beta1/agentclusterinstall_validation_hook.go

```diff
@@ -334,11 +359,70 @@ func isUserManagedNetworkingSetToFalseWithSNO(newObject *hiveext.AgentClusterIns
 		!*newObject.Spec.Networking.UserManagedNetworking
 }
 
+func validateCreatePlatformAndUMN(newObject *hiveext.AgentClusterInstall) error {
+	platform := common.PlatformTypeToPlatform(newObject.Spec.PlatformType)
+	_, _, err := provider.GetActualCreateClusterPlatformParams(
+		platform, newObject.Spec.Networking.UserManagedNetworking, getHighAvailabilityMode(newObject, nil), "")
+	return err
+}
+
+func validateUpdatePlatformAndUMNUpdate(oldObject, newObject *hiveext.AgentClusterInstall) error {
+	var (
+		platform              *models.Platform
+		userManagedNetworking *bool
+	)
+
+	if newObject.Spec.PlatformType != "" {
+		platform = common.PlatformTypeToPlatform(newObject.Spec.PlatformType)
+	} else {
+		platform = common.PlatformTypeToPlatform(oldObject.Spec.PlatformType)
+	}
+
+	if newObject.Spec.Networking.UserManagedNetworking != nil {
+		userManagedNetworking = newObject.Spec.Networking.UserManagedNetworking
+	} else {
+		userManagedNetworking = oldObject.Spec.Networking.UserManagedNetworking
+	}
+
+	_, _, err := provider.GetActualCreateClusterPlatformParams(
+		platform, userManagedNetworking, getHighAvailabilityMode(oldObject, newObject), "")
+	return err
+}
+
 func isSNO(newObject *hiveext.AgentClusterInstall) bool {
 	return newObject.Spec.ProvisionRequirements.ControlPlaneAgents == 1 &&
 		newObject.Spec.ProvisionRequirements.WorkerAgents == 0
 }
 
+func getHighAvailabilityMode(originalObject, updatesObject *hiveext.AgentClusterInstall) *string {
+	if originalObject == nil {
+		return swag.String("")
+	}
+
+	tmpObject := &hiveext.AgentClusterInstall{
+		Spec: hiveext.AgentClusterInstallSpec{
+			ProvisionRequirements: hiveext.ProvisionRequirements{
+				ControlPlaneAgents: originalObject.Spec.ProvisionRequirements.ControlPlaneAgents,
+				WorkerAgents:       originalObject.Spec.ProvisionRequirements.WorkerAgents,
+			},
+		},
+	}
+
+	if updatesObject != nil {
+		if tmpObject.Spec.ProvisionRequirements.ControlPlaneAgents != updatesObject.Spec.ProvisionRequirements.ControlPlaneAgents {
```

Comment:
Lines are a bit too long:) Maybe extract to vars: originalProvisionRequirements and updateProvisionRequirements

Reply:
Done